dialog/file.go

```diff
@@ -20,22 +20,72 @@ type textWidget interface {
 }
 
 type fileDialog struct {
+	file       *File
 	fileName   textWidget
 	open       *widget.Button
 	breadcrumb *widget.Box
 	files      *fyne.Container
 	fileScroll *widget.ScrollContainer
-	parent     fyne.Window
 
 	win      *widget.PopUp
 	selected *fileDialogItem
-	callback interface{}
 	dir      string
+}
+
+// File is a dialog containing a file picker for use in opening or saving files
```

Comment:
It’s confusing to speak of a “dialog” here if `File` does not implement `Dialog`.

Reply:
This probably needs discussion in the channel. I queried this too, but due to some methods (Hide and SetDismissText) not being applicable to mobile @okratitan decided we should not use it.

Reply:
Okay most recent commit is with the file dialog implementing Dialog.